core/genesis.go

```diff
@@ -459,8 +459,12 @@ func (g *Genesis) toBlockWithRoot(root common.Hash) *types.Block {
 	if g.GasLimit == 0 {
 		head.GasLimit = params.GenesisGasLimit
 	}
-	if g.Difficulty == nil && g.Mixhash == (common.Hash{}) {
```

Comment:
QQ, Why the `g.MixHash == (common.Hash{})` is a condition for setting the genesis difficulty?

Reply:
It was added in the PR that also added the RANDOM opcode. https://github.com/ethereum/go-ethereum/pull/24141/files#diff-1272116434e5549f278a940767af52ed2d1abc91c44fa04ec48d8835a3774036

@MariusVanDerWijden might know more.

Reply:
Yes I think that was it. It fixed a test case from the testing team if I remember correctly